### ISSUE TITLE
Localization of command and option description annotations.

### ISSUE THREAD

**Issue by poison2k** [bilal-fazlani/commanddotnet#352]
Hello, 
is there a proper way to localize the description tags of commands and options?

I localized the rest of my .NetCore console application with the ILocalizer. 

Best regards

**Comment by drewburlingame** [comment]
Localization has not been implemented. We except PRs 😀

If localizing help output is good enough, I believe you can work around this by setting the relevant attributes to the localization keys and using a [custom help provider](https://commanddotnet.bilal-fazlani.com/help/help/#custom-help-provider) to convert to Kerry to the localized text.  If you do this,  consider submitting the provider in a PR for us. 

There are still a few errors the framework will output in English and they will need to be handled separately.  We can keep this issue around for that work.

**Comment by drewburlingame** [comment]
When override the [HelpTextProvider](https://github.com/bilal-fazlani/commanddotnet/blob/master/CommandDotNet/Help/HelpTextProvider.cs#L204) for this, if you're just localizing Description and ExtendedHelpText, then you'll only need to override the methods: [ExtendedHelpTex(Command command)](https://github.com/bilal-fazlani/commanddotnet/blob/master/CommandDotNet/Help/HelpTextProvider.cs#L97) and [ArgumentDescription<T>(T argument)](https://github.com/bilal-fazlani/commanddotnet/blob/master/CommandDotNet/Help/HelpTextProvider.cs#L204)

**Comment by drewburlingame** [comment]
Looks good.  Did you try subclassing the HelpTextProvider and only overriding the properties you needed?

**Comment by drewburlingame** [comment]
Ah, thank you, this is good feedback.  I'll look at making those available.

**Comment by poison2k** [comment]
Thank you, 
and I will check if I can create a PR next week, if you see in my approach. I prefer to set string keys in a constant class and replace the hard-coded strings with this keys. This is for me best practice to work with resx files, because you have a centralized class with all keys you need to create a resx file.  Then you don't need to search through the whole source code in the hope that you have found all needed resx keys.

**Comment by drewburlingame** [comment]
This approach won't work for values assigned in attributes. In that case, the constants file seems like the best approach.

**Comment by drewburlingame** [comment]
Are you not using resx 9th the IStringLocalizer?  If not, what are you
using?

With the IStringLocalizer approach, what if we use extension methods
instead?

`localizer.Help_Command();`

With this approach, it's also clear when values will be interpolated into
the string and what they are.

**Comment by poison2k** [comment]
Hi sorry, that I not reply directly, big workload ....
Your solution sounds good, I will test your approch as soon as possible.

**Comment by drewburlingame** [comment]
@poison2k I've released a beta of localization in the latest version.  I've included the package for FluentValidation and DataAnnotations.  https://commanddotnet.bilal-fazlani.com/localization/overview/

Give it a spin and let me know what you think.  There are still strings to localize but this should cover most cases.

I want to add a package to provide resx files and another for json, and tooling to identify missing entries in those files.
